### PR TITLE
Artificial spell delay

### PR DESCRIPTION
**Pull request by Lihvar**
Adjustable artificial spell delay/lag for preventing "stuck" casting animation

Configurable via config

Related to #104

### CODE REVIEW

**Comment by MaxtorCoder** [comment]
Create a new branch and then make a pull request, also use Thread.Sleep by including System.Threading

**Review by MaxtorCoder — CHANGES_REQUESTED** [review]
Create a new branch and then make a pull request, also use Thread.Sleep by including System.Threading

**Comment by 0blu** [comment]
We can merge squash it, so there is only one commit at the end.

File: HermesProxy/World/Client/PacketHandlers/SpellHandler.cs [hunk 1]
```diff
@@ -774,70 +822,70 @@ void HandleSpellPeriodicAuraLog(WorldPacket packet)
...
-                    }
+                        {
+                            SpellPeriodicAuraLog.SpellLogEffect effect = new();
+                            effect.Effect = (uint)aura;
+                            effect.SchoolMaskOrPower = packet.ReadUInt32();
+                            effect.Amount = packet.ReadInt32();
+                            packet.ReadFloat(); // Gain multiplier
+                            spell.Effects.Add(effect);
+                            break;
+                        }
```
Comment (0blu) [inline HermesProxy/World/Client/PacketHandlers/SpellHandler.cs:888]: The indents were already correct before.

File: HermesProxy/World/Client/PacketHandlers/SpellHandler.cs [hunk 2]
```diff
@@ -126,6 +126,13 @@ void HandleUnlearnedSpells(WorldPacket packet)
         [PacketHandler(Opcode.SMSG_CAST_FAILED)]
         void HandleCastFailed(WorldPacket packet)
         {
+            // Artificial lag is needed for spell packets,
+            // or spells will bug out and glow if spammed.
+            if (Settings.ClientSpellDelay > 0)
+            {
+                Thread.Sleep(Settings.ClientSpellDelay);
+            }
```
Comment (0blu) [inline HermesProxy/World/Client/PacketHandlers/SpellHandler.cs:134]: Make just in two lines to reduce the total number of lines in each file
```cs
if (Settings.ClientSpellDelay > 0)
    Thread.Sleep(Settings.ClientSpellDelay);
```
(Applicable for all other delays)

**Review by 0blu — CHANGES_REQUESTED** [review]

**Pull request merged**